xbmc/XBDateTime.cpp

```diff
@@ -1106,6 +1106,69 @@ bool CDateTime::SetFromRFC1123DateTime(const std::string &dateTime)
   return SetDateTime(year, month, day, hour, min, sec);
 }
 
+CDateTime CDateTime::FromDateString(const std::string &date)
+{
+  CDateTime dt;
+  dt.SetFromDateString(date);
```

Comment:
`SetFromDateString` returns bool, which is rather pointless since there's already `IsValid` for checking validity. It's for constructing DateTime in one line without the unnecessary verbosity. Hopefully people will use it so we can make this class immutable and remove the setter..